### DISCUSSION TITLE
[HxGrid] Edit and Save buttons for inline editing

### DISCUSSION THREAD

**Discussion by ogbetua** [havit/Havit.Blazor#782] [Q&A] [answered]
I have added an Edit button to set the SelectedDataItem and this works really well. I have a Save button to save the changes in a row, but I don't know how to get the SelectedDataItemChanged event to fire. Any help would be appreciated.

**Comment by hakenr** [comment]
`SelectedDataItemChanged` fires when the grid changes the value of `SelectedDataItem` parameter (eg. when you click another row; when you change the page displayed etc.). It does not fire when the `SelectedDataItem` parameter value changes externaly (eg. when you change the value of field bound to `SelectedDataItem`).

**Reply by ogbetua** [reply]
I kind of figured that is what is happening. So, I can assume that this is not going to work as we need it to since we are not going to make a user click on another row just to be able to click on the Save button on the edited row.

I'll guess have to forego the inline editing.

**Reply by hakenr** [reply, marked as answer]
I'm not certain about your code structure or the specific issues you're encountering.
If you're utilizing an explicit _Save_ button, you should not be using the `HandleSelectedDataItemChanged` handler, but instead you save changes in the `HandleSaveClick` method and also set the corresponding field `selectedDataItem = null` within that method.

**Reply by ogbetua** [reply]
I think I get what you are talking about.  I was thinking about it the wrong way by expecting that event to fire. I will let you know if it works out.

**Reply by ogbetua** [reply]
Thank you for your help. I was able to get the Edit and Save buttons to work as expected
